src/shipper.py

```diff
@@ -26,34 +32,61 @@ class UnknownURL(Exception):
 
 
 class LogzioShipper(object):
-    MAX_BULK_SIZE_IN_BYTES = 1 * 1024 * 1024
+    MAX_BULK_SIZE_IN_BYTES = 3 * 1024 * 1024
 
     def __init__(self, logzio_url):
-        self._size = 0
-        self._logs = []
+        self._decompress_size = 0
+        self._logs = StringIO.StringIO()
         self._logzio_url = logzio_url
+        self._logs_counter = 0
+        try:
+            self._compress = os.environ['COMPRESS'].lower() in ["yes", "true", "t", "1"]
```

Comment:
this is weird 😄  
check what is the standard way to do it or just check not equal None

Reply:
also use .get() to not get a KeyError